boto3/s3/inject.py

```diff
@@ -356,3 +367,267 @@ def object_copy(self, CopySource, ExtraArgs=None, Callback=None,
         CopySource=CopySource, Bucket=self.bucket_name, Key=self.key,
         ExtraArgs=ExtraArgs, Callback=Callback, SourceClient=SourceClient,
         Config=Config)
+
+
+def upload_fileobj(self, Fileobj, Bucket, Key, ExtraArgs=None,
+                   Callback=None, Config=None):
+    """Upload a file-like object to S3.
+
+    This is a managed transfer which will perform a multipart upload in
+    multiple threads if necessary.
+
+    Usage::
+
+        import io
```

Comment:
The intent was to show a non-file object being uploaded, but I agree it could make adding such a method in the future be problematic.